src/data/challenge.ts

```diff
@@ -340,6 +344,15 @@ export abstract class Challenge {
     return false;
   }
 
+  /**
+   * An apply function for TRICK_ROOM challenges. Derived classes should alter this.
+   * @param isTrickRoom {@link Utils.BooleanHolder} Whether the battle is under the effect of Trick Room.
+   * @returns `true` if this function did anything.
+   */
+  applyTrickRoom(isTrickRoom: Utils.BooleanHolder): boolean {
```

Comment:
NIT:
Unused params should be prefixed with `_`
```suggestion
  applyTrickRoom(_isTrickRoom: Utils.BooleanHolder): boolean {
```